res/controllers/Denon Prime 4.midi.xml

```diff
@@ -0,0 +1,1735 @@
+              <status>0x8F</status>
+              <midino>0x01</midino>
+              <options>
+                <Script-Binding/>
+              </options>
+            </control>
+            <control> <!-- Load to Right Deck Button -->
+              <group></group>
+              <key>Prime4.rightDeck.deckLoad.input</key>
+              <description>MIDI Learned from 210 messages.</description>
+              <status>0x8F</status>
+              <midino>0x02</midino>
+              <options>
+                <Script-Binding/>
+              </options>
+            </control>
+			<control> <!-- VIEW Button -->
+				<group>[Master]</group>
+				<key>Prime4.maxView.input</key>
+				<description>Maximize Mixxx Library View</description>
+				<status>0x9F</status>
+				<midino>0x07</midino>
+				<options>
+					<Script-Binding/>
+				</options>
+			</control>
+			<control>
+				<group>[Master]</group>
+				<key>Prime4.maxView.input</key>
+				<description>Maximize Mixxx Library View</description>
+				<status>0x8F</status>
+				<midino>0x07</midino>
+				<options>
+					<Script-Binding/>
+				</options>
+			</control>
+            <control> <!-- Headphone Gain Knob -->
+              <group>[Master]</group>
+              <key>headGain</key>
+              <description>MIDI Learned from 408 messages.</description>
+              <status>0xBF</status>
+              <midino>0x0A</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
+            <control> <!-- Headphone PFL/Master Mix Knob -->
+              <group>[Master]</group>
+              <key>headMix</key>
+              <description>MIDI Learned from 390 messages.</description>
+              <status>0xBF</status>
+              <midino>0x09</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
+            <control> <!-- Headphone Split Button -->
+              <group>[Master]</group>
+              <key>headSplit</key>
+              <description>MIDI Learned from 28 messages.</description>
+              <status>0x9F</status>
+              <midino>0x0B</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
+            <control> <!-- Crossfader -->
+              <group>[Master]</group>
+              <key>crossfader</key>
+              <description>MIDI Learned from 1089 messages.</description>
+              <status>0xBF</status>
+              <midino>0x0E</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
+
+	    <control> <!-- Deck 1 - Crossfader Assign Switch -->
+	      <group>[Channel1]</group>
+	      <key></key>
+	      <description>Entered manually with Emacs</description>
+	      <status>0x90</status>
+	      <midino>0x0F</midino>
+	      <options>
+		<Script-Binding/>
+	      </options>
+	    </control>
```

Comment:
is this a placeholder? The `key` is empty.

Reply:
Yeah, I'm currently working on this now. The crossfader assignment switches on the Prime 4 output value 0 when switched to the left, value 1 in the centre, and value 2 on the right. Just trying to figure out how to handle that inside ComponentsJS

Reply:
Yes, this is a placeholder. I'm trying to write a function that handles the console's switches properly, but am not sure how yet. They work the same way the internal ones in Mixxx work: value 0 for left, 1 for centre, and 2 for right. Just not sure how to make a Component handle this behaviour.

Reply:
Have you tried something like this:
```
new components.Button({
    midi: [0x90, 0x0F],
    inKey: "orientation"
    input: function(_channel, _control, value, _status, _group) {
        this.inSetValue(value);
    }
});
```

Reply:
I think I tried something similar but kept getting errors. I'll give this a try when I next am free 👍

Reply:
Please do, if you still encounter error, please post them so I can help you diagnose the issue.